scripts/main.js

```diff
@@ -0,0 +1,51 @@
+class Hero {
+    constructor(weapons) {
+      this.weapons = weapons;
+    }
+    
+    HitBySword() {
+      if(this.weapons.sword != null) {
```

Comment:
There is possibility to get undefined or null. So you can just declare 
if(this.weapons.sword) and it will be more safely, because if there is weapons without sword property this.weapons.sword will contain undefined, consiquently Hit() method doesn't exist, but you will try to use him and get an error

Reply:
So, in this case everything is fine, but try to keep in mind my comment and use strict equality in future (=== !==)

Reply:
Got it. Fixed